tests/hpc/slurm_master_db.pm

```diff
@@ -52,9 +52,7 @@ sub run {
     barrier_wait("SLURM_MASTER_SERVICE_ENABLED");
     barrier_wait("SLURM_SLAVE_SERVICE_ENABLED");
 
-    # run basic test against first compute node
-    # add sleep; bugzilla#1140403
-    sleep(5);
+    $self->check_nodes_availability();
```

Comment:
Where is `check_nodes_availability` defined? Missing files in commit?

Reply:
@czerw in the lib/hpcbase.pm
That was merged earlier on for experimental testing.

https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/5a66ad15a6743d38ea82a32ae9c041385ea565f5#diff-202b6b62531ad5ac248280ffc8c75598

Reply:
@czerw I was thinking about your comment regarding that sleep() and then ended up having this generic check if nodes can be 'accessible'. I'm starting with ping, but this approach can be used in even more generic and fancy way ... I think.

Reply:
Ahh, quite new change, which i was not aware about :)

Reply:
But you are basically checking network availability using ping(emulating sleep with 5 ping counts), not that slurm service is ready. This can be misleading. This approach will scale with number of nodes,  5 compute nodes will lead to 25 ping counts in total, but you can't be 100% sure that slurm is in ready state on all of them.

Reply:
Sure, valid point. I meant to only check for now if in fact nodes could be accessed.
The idea behind introducing that nasty sleep was to have some more time for the VMs to get ready. Strictly speaking neither me nor HPC devs do not know what is the root cause of that bug (which manifests in srun jobs not being scheduled), which might be - as I suspect it - a problem with limited resources of our VMs which we pretend to be an hpc cluster.
So this simple ping is really exactly meant as only checking if nodes are 'pinggable', so problems as "srun cannot be scheduled" can be easier isolated and thus investigated.

So yeah, this isn't about slurm availability;  this is about some more granular checks for the cluster to isolate potential problems.

Does it make sens to you?